VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitWebviewFragment.java

```diff
@@ -101,67 +118,114 @@ public void onActivityCreated(Bundle savedInstanceState) {
+                                                    WebResourceResponse errorResponse) {
+                        didReceiveError = true;
+                        switch (errorResponse.getStatusCode()) {
+                            case HttpURLConnection.HTTP_FORBIDDEN:
+                            case HttpURLConnection.HTTP_UNAUTHORIZED:
+                            case HttpURLConnection.HTTP_NOT_FOUND:
+                                EdxCookieManager.getSharedInstance().tryToRefreshSessionCookie();
+                                break;
+                        }
+                        showErrorMessage(R.string.network_error_message,
+                                FontAwesomeIcons.fa_exclamation_circle);
+                    }
+
+                    public void onPageFinished(WebView view, String url) {
+                        if (didReceiveError) {
+                            didReceiveError = false;
+                            return;
+                        }
+                        if (url != null && url.equals("data:text/html," + EMPTY_HTML)) {
+                            //we load a local empty html page to release the memory
+                        } else {
+                            pageIsLoaded = true;
+                        }
+
+                        //TODO -disable it for now. as it causes some issues for assessment
+                        //webview to fit in the screen. But we still need it to show additional
+                        //compenent below the webview in the future?
+                        // view.loadUrl("javascript:EdxAssessmentView.resize(document.body.getBoundingClientRect().height)");
+                        ViewPagerDownloadManager.instance.done(CourseUnitWebviewFragment.this, true);
+                        hideLoadingProgress();
+                    }
+                };
         client.setAllLinksAsExternal(true);
         //webView.addJavascriptInterface(this, "EdxAssessmentView");
 
         if(  ViewPagerDownloadManager.USING_UI_PRELOADING ) {
-            if (ViewPagerDownloadManager.instance.inInitialPhase(unit))
+            if (ViewPagerDownloadManager.instance.inInitialPhase(unit)) {
                 ViewPagerDownloadManager.instance.addTask(this);
-            else
+            } else {
                 tryToLoadWebView(true);
+            }
+        }
+    }
+
+    /**
+     * Shows the error message with the given icon, if the web page failed to load
+     * @param errorMsg The error message to show
+     * @param errorIcon The error icon to show with the error message
+     */
+    private void showErrorMessage(@StringRes int errorMsg, @NonNull Icon errorIcon) {
+        if (!pageIsLoaded) {
+            errorTextView.setVisibility(View.VISIBLE);
+            errorTextView.setText(errorMsg);
+            TextViewCompat.setCompoundDrawablesRelativeWithIntrinsicBounds(errorTextView, null,
```

Comment:
Since the drawable is applied to the top, we don't need to use the "relative" method here, or `TextViewCompat`.